### PR TITLE
Validate uniqueness of custom primary key

### PR DESCRIPTION
**Pull request by presskey**
There was a problem in uniqueness validator that allowed model with custom primary key to pass validation when non unique primary key value was set which resulted in exception.

### CODE REVIEW

File: activerecord/test/cases/validations/uniqueness_validation_test.rb [hunk 1]
```diff
@@ -427,4 +428,18 @@ def test_does_not_validate_uniqueness_of_if_parent_record_is_validate_false
...
+
+    key1 = Keyboard.new(key_number: 10)
+    assert key1.save, "Should save key1 as unique"
+
+    key2 = Keyboard.new(key_number: 11)
+    assert key2.save, "Should save key2 as unique"
+
+    key2.key_number = 10
+    assert !key2.valid?, "Shouldn't be valid"
+    assert !key2.save, "Shouldn't save since primary key is't unique"
```
Comment (repinel) [inline activerecord/test/cases/validations/uniqueness_validation_test.rb]: `is't` to `it's`

Comment (presskey) [inline activerecord/test/cases/validations/uniqueness_validation_test.rb]: @repinel thank you. I've replaced ``is't`` with ``isn't``

Comment (repinel) [inline activerecord/test/cases/validations/uniqueness_validation_test.rb]: Cool :smile:

File: activerecord/lib/active_record/validations/uniqueness.rb [hunk 2]
```diff
@@ -17,7 +17,12 @@ def validate_each(record, attribute, value)
         value = map_enum_attribute(finder_class, attribute, value)
 
         relation = build_relation(finder_class, table, attribute, value)
-        relation = relation.where.not(finder_class.primary_key => record.id) if record.persisted?
+
+        if record.persisted?
+          primary_key = finder_class.primary_key.to_sym
```
Comment (sgrif) [inline activerecord/lib/active_record/validations/uniqueness.rb]: These shouldn't be symbolized.

File: activerecord/test/cases/validations/uniqueness_validation_test.rb [hunk 3]
```diff
@@ -427,4 +428,18 @@ def test_does_not_validate_uniqueness_of_if_parent_record_is_validate_false
     assert reply.valid?
     assert topic.valid?
   end
+
+  def test_validate_uniqueness_of_custom_primary_key
+    Keyboard.validates_uniqueness_of(:key_number)
```
Comment (sgrif) [inline activerecord/test/cases/validations/uniqueness_validation_test.rb]: This test will leak state, as you've modified the class. Please create an anonymous class for use in the test.

**Comment by sgrif** [comment]
I went ahead and fixed the issues I had with the implementation, added a changelog, and committed. You've been credited for the commit. https://github.com/rails/rails/commit/119b9181ece399c67213543fb5227b82688b536f

**Pull request closed without merging**